iothub/device/tests/ClientPropertyCollectionTestsNewtonsoft.cs

```diff
@@ -0,0 +1,84 @@
+    {
+        private const string propertyName = "propertyOne";
+        private const string simplePropertyValue = "propertyValue";
+        private const string componentName = "testableComponent";
+        private const string writablePropertyDescription = "testableWritablePropertyDescription";
+        private const string updatedPropertyValue = "updatedPropertyValue";
+        
+        private const string twinJson = "{ \"propertyOne\" : \"propertyValue\" }";
+        private const string twinJsonWritableProperty = "{ \"propertyOne\" : { \"value\" : \"propertyValue\",  \"ac\" : 200, \"av\" : 2, \"ad\" : \"testableWritablePropertyDescription\" } }";
+        private const string twinJsonWithComponent = "{ \"testableComponent\" : { \"__t\" : \"c\",  \"propertyOne\" : \"propertyValue\" } }";
+        private const string twinJsonWritablePropertyWithComponent = "{ \"testableComponent\" : { \"__t\" : \"c\", \"propertyOne\" : { \"value\" : \"propertyValue\",  \"ac\" : 200, \"av\" : 2, \"ad\" : \"testableWritablePropertyDescription\" } } }";
+
+        private static TwinCollection collectionToRoundTrip = new TwinCollection(twinJson);
+        private static TwinCollection collectionWritablePropertyToRoundTrip = new TwinCollection(twinJsonWritableProperty);
+        private static TwinCollection collectionWithComponentToRoundTrip = new TwinCollection(twinJsonWithComponent);
+        private static TwinCollection collectionWritablePropertyWithComponentToRoundTrip = new TwinCollection(twinJsonWritablePropertyWithComponent);
+
+        [TestMethod]
+        public void ClientPropertyCollectionNewtonsoft_CanGetSimpleValue()
+        {
+            
+            var clientProperties = ClientPropertyCollection.FromTwinCollection(collectionToRoundTrip, DefaultPayloadConvention.Instance);
+            clientProperties.Convention = DefaultPayloadConvention.Instance;
+            clientProperties.TryGetValue<string>(propertyName, out var outValue);
+            Assert.AreEqual(simplePropertyValue, outValue, $"Property values do not match, expected {simplePropertyValue} but got {outValue}");
+        }
+
+        [TestMethod]
+        public void ClientPropertyCollectionNewtonsoft_CanGetSimpleValueWithComponent()
+        {
+            var clientProperties = ClientPropertyCollection.FromTwinCollection(collectionWithComponentToRoundTrip, DefaultPayloadConvention.Instance);
+            clientProperties.TryGetValue<string>(componentName, propertyName, out var outValue);
+            Assert.AreEqual(simplePropertyValue, outValue, $"Property values do not match, expected {simplePropertyValue} but got {outValue}");
+        }
+
+        [TestMethod]
+        public void ClientPropertyCollectionNewtonsoft_CanAddSimpleWritablePropertyAndGetBackWithoutDeviceClient()
+        {
+            var clientProperties = ClientPropertyCollection.FromTwinCollection(collectionWritablePropertyToRoundTrip, DefaultPayloadConvention.Instance);
```

Comment:
this shouldn't be using twin collection, right? this is user created collection, isn't it?

Reply:
The idea here is we're testing the flow coming off of the desired property callback for SubscribeToWritablePropertiesEventAsync. Tthis will test the deserialization of the JSON object and our ability to get the values.

We also need to test ToClientProperties to cover the MqttTransport (and others),

Reply:
I see, the test method name needs a renaming then, right? (right now it says "...WithoutDeviceClient"), whereas if we're testing the writable property callback flow then both device client and the convention would already be set up.